src/registrar/models/models.py

```diff
@@ -0,0 +1,422 @@
+
+
+class Territory(models.Model):
+    # TODO: ability to populate with census data
+    name = models.CharField(max_length=2)
+
+
+class FederalAgency(models.Model):
+    # TODO: ability to populate with data from ???
+    name = models.CharField(max_length=2)
+
+
+class GovEntity(TimeStampedModel):
+    # TODO: choices ????
+    entity_type = models.CharField(max_length=2)
+    fallback_contact_instructions = models.CharField(max_length=2)
+    city = models.ManyToManyField(City)
+    state = models.ManyToManyField(State)
+    county = models.ManyToManyField(County)
+    tribe = models.ManyToManyField(Tribe)
+    territory = models.ManyToManyField(Territory)
+    agency = models.ManyToManyField(FederalAgency)
+
+
+
+class AuthorizingOfficial(TimeStampedModel, ContactModel, AddressModel):
+    # contact's name
+    name = models.CharField(max_length=2)
+    # alternative contacting instructions (could also be foreign key to AO record)
+    alt_contact_info = models.CharField(max_length=2)
+    # an AO may or may not have a user in the system
+    user = models.ForeignKey(User, on_delete=models.SET_NULL, null=True)
+
+
+class Organization(TimeStampedModel, AddressModel):
+    name = models.CharField(max_length=2)
+    entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+
+
+class Rejection(models.Model):
+    reason = models.CharField(max_length=2)
+
+
+class Application(TimeStampedModel):
+    # user who is making the request
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL)
+    # the person who says this user is allowed to request this domain
+    authorizing_official = models.ForeignKey(AuthorizingOfficial, on_delete=models.CASCADE)
+    # the government who CISA says is allowed to request a .gov domain
+    authorizing_entity = models.ForeignKey(GovEntity, on_delete=models.CASCADE)
+    # the subdivision or department this user belongs to
+    organization = models.ForeignKey(Organization, on_delete=models.PROTECT)
+    domain_name = models.CharField(max_length=2)
+    is_exception_requested = models.BooleanField()
+    # if the user wants an ineligible domain name, they may give a reason for their request
+    exception_rationale = models.CharField(max_length=2)
+    is_expedited_requested = models.BooleanField()
+    # if the user wants a fast response, they may give a reason for their request
+    expedited_rationale = models.CharField(max_length=2)
+    hsts_acknowledged = models.BooleanField()
+    # reason for requesting the domain
+    purpose = models.CharField(max_length=2)
+    # Choices ??? -- also, TODO: use a state machine to model the business logic
+    status = models.CharField(max_length=2)
+    # choices ???
+    rejection_codes = models.ManyToManyField(Rejection)
+    assignee = models.ForeignKey(User, null=True, on_delete=models.SET_NULL, related_name="assigned")
```

Comment:
Would lean in favor of "let's design for the future" -- from the analyst interviews it is pretty clear that passing applications back and forth is a normal part of their workflow, so it would be great to have the ability to assign to another user (for review or comment, etc) without losing track of one's own tickets.

This may need to be refined in some way as we continue (like `original_assignee`, `approved_by`, etc... those would all be in the audit logs, but if the system has to find who those users are, it would perhaps be good to cache them on the model).